packages/block-editor/src/components/focus-capture/index.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * External dependencies
+ */
+import { last } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { focus } from '@wordpress/dom';
+
+/**
+ * Internal dependencies
+ */
+import { getBlockFocusableWrapper } from '../../utils/dom';
+
+function selector( select ) {
+	const {
+		isNavigationMode,
+		getBlockSelectionStart,
+	} = select( 'core/block-editor' );
+	return {
+		clientId: getBlockSelectionStart(),
+		isNavigationMode: isNavigationMode(),
+	};
+}
+
+function FocusCaptureElement( { reverse } ) {
+	const { clientId, isNavigationMode } = useSelect( selector, [] );
+
+	function onFocus() {
+		const wrapper = getBlockFocusableWrapper( clientId );
+
+		if ( reverse ) {
+			const tabbables = focus.tabbable.find( wrapper );
+			last( tabbables ).focus();
+		} else {
+			wrapper.focus();
+		}
+	}
+
+	return (
+		<div
+			tabIndex={ clientId && ! isNavigationMode ? '0' : undefined }
+			onFocus={ onFocus }
+			// Needs to be positioned within the viewport, so focus to this
+			// element does not scroll the page.
+			style={ { position: 'fixed' } }
```

Comment:
I feel this would scroll the page if it's parent is not fixed?

Reply:
Which parent? The parent positioning doesn't affect the position of a fixed child.

Reply:
Mmm, yeah maybe I'm not thinking properly here.